test/string_view-test/string_view-test.cpp

```diff
@@ -0,0 +1,390 @@
+	//where rcount is the smaller of count and size() - pos.
+	size_t len_expected = (n < sv_f.size() - pos) ? n : sv_f.size() - pos;
+	is_true(rlen == len_expected, "sv_f.copy(z_copy, n, pos)");
+	is_zero(strcmp(z_copy, z_copy_expected), "z_copy == z_copy_expected");
+	try {
+		//throw std::out_of_range exception if pos > size() 
+		sv_f.copy(z_copy, n, sv_f.size() + 1);
+		is_true(false, "sv_f.copy(z_copy, n, pos), should not be executed.");
+	}
+	catch (const std::out_of_range&) {
+		is_true(true, "if pos > size(), then throw out_of_range exception");
+	}
+	catch (...) {
+		is_true(false, "Throw an unexpected exception.");
+	}
+
+	//substr
+	sv = "something";
+	pos = 4;
+	n = 5;
+	string_view sub_expected{ "thing" };
+
+	string_view sub_sv{ sv.substr(pos, n) };
+	//substr(size_type pos = 0, size_type count = npos)
+	//Returns a view of the substring [pos, pos + rcount), 
+	//where rcount is the smaller of count and size() - pos
+	len_expected = (n < sv.size() - pos) ? n : sv.size() - pos;
+	is_true(sub_sv.size() == len_expected, "sv.copy(pos, n).size");
+	is_true(sub_sv.data() == sv.data() + pos, "sv.copy(pos, n).data");
+	is_true(sub_sv == sub_expected, "");
+	try {
+		//throw std::out_of_range exception if pos > size() 
+		pos = sv.size() + 1;
+		sub_sv = sv.substr(pos, n);
+		is_true(false, "sv.substr(pos, n), should not be executed.");
+	}
+	catch (const std::out_of_range&) {
+		is_true(true, "if pos > size(), then throw out_of_range exception");
+	}
+	catch (...) {
+		is_true(false, "sv.substr(pos, n), throw an unexpected exception.");
+	}
+
+	//compare 
+	string_view sv_a_to_e{ "abcde" };
+	string_view sv_A_to_E{ "ABCDE" };
+	string_view sv_a_to_c{ "abc" };
+
+	is_positive(sv_a_to_e.compare(sv_A_to_E), "sv_a_to_e.compare(sv_A_to_E)");
+	is_negative(sv_A_to_E.compare(sv_a_to_e), "sv_A_to_E.compare(sv_a_to_e)");
+	is_zero(sv_a_to_e.compare(0, 3, sv_a_to_c), "sv_a_to_e.compare(0, 3, sv_A_to_E)");
+	is_negative(sv_A_to_E.compare(0, 3, sv_a_to_c), "sv_a_to_e.compare(0, 3, sv_A_to_E)");
+	is_negative(sv_a_to_c.compare(sv_a_to_e), "sv_a_to_c.compare(sv_a_to_e) < 0");
+	is_positive(sv_a_to_c.compare(sv_A_to_E), "sv_a_to_c.compare(sv_A_to_E) > 0");
+	is_positive(sv_a_to_e.compare(1, 1, sv_A_to_E, 1, 1), "sv_a_to_e.compare(1, 1, sv_A_to_E, 1, 1)");
+	is_negative(sv_A_to_E.compare(1, 1, sv_a_to_c, 1, 1), "sv_A_to_E.compare(1, 1, sv_a_to_c, 1, 1)");
+	is_zero(sv_a_to_c.compare(1, 1, sv_a_to_e, 1, 1), "sv_a_to_c.compare(1, 1, sv_a_to_e, 1, 1)");
+
+	//compare empty string_view
+	is_negative(sv_empty.compare(sv_a_to_e), "sv_empty.compare(sv_a_to_e)");
+	is_positive(sv_a_to_c.compare(sv_empty), "sv_a_to_c.compare(sv_empty)");
+
+	const char* a_ABC{ "ABC" };
+	is_positive(sv_a_to_e.compare(a_ABC), "sv_a_to_e.compare(str)");
+	is_zero(sv_A_to_E.compare(0, 3, a_ABC), "sv_empty.compare(0, 5, str)");
+	is_positive(sv_a_to_c.compare(a_ABC), "sv_empty.compare(0, 5, str, 5)");
+
+
+	//8. Searching
+	//    0         1 
+	//    01234567890123456
```

Comment:
Perhaps I'm missing something but what are these comments for?

Reply:
I fixed it. [see @smurthys's comment ](https://github.com/sigcpp/stl-lite/pull/23#commitcomment-38708357)

Reply:
Ahh OK that makes sense. It is hard to know what those numbers are for without seeing that comment.